infra/base-images/base-builder/detect_repo.py

```diff
@@ -29,25 +29,37 @@
 
 
 def main():
-  """Function to get a git repos information based on its commit."""
+  """Function to get a git repos information based on its commit.
+
+  Raises:
+    ValueError when a commit or a ref is not provided
+  """
   parser = argparse.ArgumentParser(
       description='Finds a specific git repo in an oss-fuzz projects docker file.'
   )
   parser.add_argument(
       '--src_dir',
       help='The location of the oss-fuzz projects source directory',
       required=True)
+  parser.add_argument(
+      '--repo_name', help='the name of the github repo', required=False)
   parser.add_argument(
       '--example_commit',
       help='A commit SHA refrencing the projects main repo',
-      required=True)
+      required=False)
+
   args = parser.parse_args()
+  if not args.repo_name and not args.example_commit:
+    raise ValueError(
+        'Requires either a example commit or a repo name to detect repo.')
   for single_dir in os.listdir(args.src_dir):
     full_path = os.path.join(args.src_dir, single_dir)
     if not os.path.isdir(full_path):
       continue
-    if check_for_commit(
-        os.path.join(args.src_dir, full_path), args.example_commit):
+    if args.example_commit and check_for_commit(full_path, args.example_commit):
+      print('Detected repo: %s %s' % (get_repo(full_path), single_dir.rstrip()))
```

Comment:
No need for string formatting print(x, y, z) works. Please change all uses of print that this CL introduces to do this (including `print('blah ' + x)`)

Reply:
I'm confused on why to avoid string formatting. In this example I need the space for the regex so the print statement would turn into print('Detected repo: ' + get_repo(full_path) + ' ' + single_dir) which seems messier .

Reply:
What you have currently takes more effort to understand than what I asked for:
`print('Detected repo:', get_repo(full_path), single_dir.rstrip())`
That is why you should prefer using it.

Reply:
Also what is the purpose of `rstrip` here?
rstrip seems to be used a lot in your code and I wonder if it's actually necessary most of the time.

Reply:
Done.